app/controllers/peoplefinder/tokens_controller.rb

```diff
@@ -16,10 +16,16 @@ def create
 
     def show
       token = Token.where(value: params[:id]).first
-      return forbidden unless token
-
-      person = FindCreatePerson.from_token(token)
-      login_person(person)
+      if token && token.active?
+        person = FindCreatePerson.from_token(token)
+        login_person(person)
+      elsif token.nil?
+        error :invalid_token
+        redirect_to new_sessions_path
+      elsif !token.active?
+        error :expired_token, hours: Token.ttl
+        redirect_to new_sessions_path
+      end
```

Comment:
This logic seems a bit convoluted, because we're testing `active?` twice. I wonder if this might be better:

```ruby
token = Token.where(value: params[:id]).first
if token
  attempt_token_login token
else
  reject_missing_token
end
```

where `attempt_token_login` is something like:

```ruby
if token.active?
  person = FindCreatePerson.from_token(token)
  login_person(person)
else
  error :expired_token, hours: Token.ttl
  redirect_to new_sessions_path
end
```